### PR TITLE
[KYUUBI #5035] Spark engine session page display session end time and duration

### LINKED ISSUE #5035: [FEATURE] Spark engine session page display session end time and duration
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature
Now the engine session page only has session start time.
```
kyuubi/session/?id=foo
```

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by lsm1**
### _Why are the changes needed?_
close #5035 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image
- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
Any screenshots for comparing the changes?

**Comment by lsm1** [comment]
> Any screenshots for comparing the changes?
ok，like this
image

**Review by cxzl25 — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineSessionPage.scala [hunk 1]
```diff
@@ -80,6 +80,16 @@ case class EngineSessionPage(parent: EngineTab)
           IP {sessionStat.ip},
           Server {sessionStat.serverIp},
           Session created at {formatDate(sessionStat.startTime)},
+          {
+          if (sessionStat.endTime == -1L) {
```
Comment (bowenliang123) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineSessionPage.scala:84]: Or just <0?

**Review by bowenliang123 — APPROVED** [review]
lGTM overall.

**Review by cfmcgrady — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineSessionPage.scala [hunk 2]
```diff
@@ -80,6 +80,16 @@ case class EngineSessionPage(parent: EngineTab)
           IP {sessionStat.ip},
           Server {sessionStat.serverIp},
           Session created at {formatDate(sessionStat.startTime)},
+          {
+          if (sessionStat.endTime < 0) {
+            ""
+          } else {
+            s"""
+               |Session end at ${formatDate(sessionStat.endTime)},
```
Comment (bowenliang123) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineSessionPage.scala:88]:
```suggestion
               | ended at ${formatDate(sessionStat.endTime)},
```

**Comment by bowenliang123** [comment]
How about changed to separated lines for connection details and the statistics.
```
        <h4>
          User {sessionStat.username},
          IP {sessionStat.ip},
          Server {sessionStat.serverIp},
        </h4> ++
        <h4>
          Session created at {formatDate(sessionStat.startTime)},
          if (sessionStat.endTime > 0) {
          s"""
             | ended at ${formatDate(sessionStat.endTime)},
             | after ${formatDuration(sessionStat.duration)}.
             |""".stripMargin
        }
          Total run {sessionStat.totalOperations} SQL
        </h4> ++
```

**Review by bowenliang123 — APPROVED** [review]

**Comment by bowenliang123** [comment]
Would you like to update the screenshots in the description? And I think this PR is good to go.  @lsm1

**Comment by lsm1** [comment]
> Would you like to update the screenshots in the description? And I think this PR is good to go. @lsm1

ok，done

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**